pkg/components/metallb/component_test.go

```diff
@@ -97,3 +107,197 @@ component "metallb" {
 `
 	testRenderManifest(t, configHCL)
 }
+
+func getSpeakerDaemonset(t *testing.T, m map[string]string) *appsv1.DaemonSet {
+	dsStr, ok := m["daemonset-speaker.yaml"]
```

Comment:
I don't like hardcoding file names here. Couldn't we try to parse all files as DaemonSet and match by name?

Reply:
How is using name not hardcoding? How often this changes?

Reply:
The DaemonSet name is less likely to change than the file name. It also puts the constraint on different logical level then (from file system level to manifest level).

Reply:
I think this is inefficient. Because each time we are converting it, then we are finding what we want. I can't pass that object around because it is not exposed. I think we should expose it and then fix these tests. I think for now we should go with just file name with trade off changing the name when it changes in the config.

```diff
diff --git a/pkg/components/metallb/component_test.go b/pkg/components/metallb/component_test.go
index c21753985..42bd25fb7 100644
--- a/pkg/components/metallb/component_test.go
+++ b/pkg/components/metallb/component_test.go
@@ -24,6 +24,7 @@ import (
        "sigs.k8s.io/yaml"
 
        "github.com/kinvolk/lokomotive/pkg/components/util"
+       "github.com/kinvolk/lokomotive/pkg/k8sutil"
 )
 
 func TestEmptyConfig(t *testing.T) {
@@ -109,8 +110,20 @@ component "metallb" {
 }
 
 func getSpeakerDaemonset(t *testing.T, m map[string]string) *appsv1.DaemonSet {
-       dsStr, ok := m["daemonset-speaker.yaml"]
-       if !ok {
+       parsedManifest, err := k8sutil.LoadManifests(m)
+       if err != nil {
+               t.Fatalf("failed to load rendered manifests: %v", err)
+       }
+
+       var dsStr string
+
+       for _, pm := range parsedManifest {
+               if pm.Kind() == "DaemonSet" && pm.Name() == "speaker" {
+                       dsStr = string(pm.Raw())
+               }
+       }
+
+       if dsStr == "" {
                t.Fatalf("speaker daemonset config not found")
        }
 
@@ -123,8 +136,20 @@ func getSpeakerDaemonset(t *testing.T, m map[string]string) *appsv1.DaemonSet {
 }
 
 func getDeployController(t *testing.T, m map[string]string) *appsv1.Deployment {
-       deployStr, ok := m["deployment-controller.yaml"]
-       if !ok {
+       parsedManifest, err := k8sutil.LoadManifests(m)
+       if err != nil {
+               t.Fatalf("failed to load rendered manifests: %v", err)
+       }
+
+       var deployStr string
+
+       for _, pm := range parsedManifest {
+               if pm.Kind() == "Deployment" && pm.Name() == "controller" {
+                       deployStr = string(pm.Raw())
+               }
+       }
+
+       if deployStr == "" {
                t.Fatalf("controller deployment config not found")
        }
```

Reply:
We discussed offline and for the time being it should be okay'ish to hardcode the names.